frontend/__tests__/unit/components/Badges.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { render, screen } from '@testing-library/react'
+import React from 'react'
+import Badges from 'components/Badges'
+
+// Mock FontAwesome
+jest.mock('@fortawesome/fontawesome-svg-core', () => ({
+  findIconDefinition: jest.fn(({ iconName }: { iconName: string }) => {
+    // Return truthy value for valid icons, null for invalid
+    return iconName === 'medal' ? { iconName } : null
+  }),
+}))
+
+jest.mock('wrappers/FontAwesomeIconWrapper', () => {
+  return function MockFontAwesomeIconWrapper({ icon }: { icon: string }) {
+    const iconName = icon.split(' ').pop()?.replace('fa-', '') || icon.replace('fa-', '')
+    return <span data-testid={`icon-${iconName}`} />
```

Comment:
Also your test does not test actual icon rendering - the mock just renders a span, not the actual icon.